relay-event-normalization/src/normalize/span/ai.rs

```diff
@@ -364,4 +378,148 @@ mod tests {
         }
         ");
     }
+
+    /// Test that the AI operation type is inferred from a gen_ai.operation.name attribute.
+    #[test]
+    fn test_infer_ai_operation_type_from_gen_ai_operation_name() {
+        let mut operation_types = HashMap::new();
+        operation_types.insert(Pattern::new("*").unwrap(), "ai_client".to_owned());
+        operation_types.insert(Pattern::new("invoke_agent").unwrap(), "agent".to_owned());
+        operation_types.insert(
+            Pattern::new("gen_ai.invoke_agent").unwrap(),
+            "agent".to_owned(),
+        );
```

Comment:
Usually nicer to construct the hashmap in one go, also means it doesn't need to be marked as mutable.

```suggestion
        let operation_types = HashMap::from([
            (Pattern::new("*").unwrap(), "ai_client".to_owned()),
            (Pattern::new("invoke_agent").unwrap(), "agent".to_owned()),
            (
                Pattern::new("gen_ai.invoke_agent").unwrap(),
                "agent".to_owned(),
            ),
        ]);
```

Reply:
Thanks, this looks better!